src/terminal/input/terminalInput.cpp

```diff
@@ -866,3 +874,45 @@ std::wstring TerminalInput::_GenerateWin32KeySequence(const KeyEvent& key)
                        key.GetActiveModifierKeys(),
                        key.GetRepeatCount());
 }
+
+// GH#1304 - On <Escape> or <Return> key input, switch IME state to alphanumeic mode.
+// - It loads "IMM32.DLL" module dynamically.
+
+static HMODULE IMMDLL = 0;
+
+static HWND (*_ImmGetDefaultIMEWnd)(HWND);
+static HIMC (*_ImmGetContext)(HWND);
+static BOOL (*_ImmGetOpenStatus)(HIMC);
+static BOOL (*_ImmSetConversionStatus)(HIMC, DWORD, DWORD);
+static BOOL (*_ImmReleaseContext)(HWND, HIMC);
+
+void TerminalInput::CheckAlphaMode(WORD key)
+{
+    // check KeyAlphaMode setting and do IME reset.
+    // if (!ServiceLocator::LocateGlobals().KeyAlphaMode() || (key != VK_ESCAPE && key != VK_RETURN))
+    // TODO: Apply globals.KeyAlphaMode setting
+    if (key != VK_ESCAPE && key != VK_RETURN)
+        return;

```

Comment:
This is for future plan to make current option setting binary on/off toggle switch to 4-way choice combo box or two check boxes: Apply both return and escape key, Escape key only, Return key only, or apply to None, depending on option setting value. `TerminalInput::HandleKey()` calls `CheckAlphaMode()` only when pressing escape or return key to save call. Currently it is redundant check. It will be combined with option setting check when it works right.